test/unit/test_str.c

```diff
@@ -42,6 +42,14 @@ bool test_rz_str_replace(void) {
 	mu_end;
 }
 
+bool test_rz_str_ncpy(void) {
+	char *str = strdup("hello world");
+	char buf[10];
+	rz_str_ncpy(buf, str, 3);
```

Comment:
please add a test for empty strings, for string of length 1, for size = 0, size=1, size=sizeof(buf), size=sizeof(buf)-1 because those may be tricky. Also, make sure to run the unit tests with an ASAN build.

Reply:
See also https://github.com/rizinorg/rizin/blob/dev/test/unit/legacy_unit/util/test_str.c#L60